pkg/bmc/access.go

```diff
@@ -93,14 +94,14 @@ func getTypeHostPort(address string) (bmcType, host, port, path string, err erro
 			path = parsedURL.Path
 		}
 	}
-	return bmcType, host, port, path, nil
+	return bmcType, host, port, path, query, nil
 }
 
 // NewAccessDetails creates an AccessDetails structure from the URL
 // for a BMC.
 func NewAccessDetails(address string) (AccessDetails, error) {
 
-	bmcType, host, port, path, err := getTypeHostPort(address)
+	bmcType, host, port, path, query, err := getTypeHostPort(address)
```

Comment:
What do you think about passing the parsed URL to the factory directly? We could replace getTypeHostPort() with a function that tries to parse a URL string, and if it can't, construct a parsed URL for itself based on the "host:port" syntax.

Reply:
that would be much cleaner indeed. I'll make the change and check the CI